File: LSE.cs

using System;
public class LSE : Lista{
    private Noh inicio;


    public LSE(){
        Noh inicio = null;
    }

    public bool estahVazia()
    {
        throw new NotImplementedException();
    }

    public void insereFim(object item)
    {
        Noh novo = new Noh(item);

        if(inicio == null){
            inicio = novo;
        }
        else{
            Noh ultimo;
            for(Noh i = inicio; i != null; i = i.Proximo){
                ultimo = i;
                ultimo.Proximo = novo;
            }
        }
    }

    public void insereInicio(object item)
    {
        Noh novo = new Noh(item);

        if(inicio == null){
            inicio = novo;
        }
        else {
            novo.Proximo = inicio;
            inicio = novo;
        }
    }

    public bool remove(object item)
    {
        
        Noh p;
        Noh ant = null;

        p = inicio;
        while(p != null && p.Item != item){
            ant = p;
            p = p.Proximo;
        }

        if(p == null ){  
            inicio = p.Proximo;
            return false;
        }
        
        if(ant == null){
            inicio = p.Proximo;
        }
        
        else{
            ant.Proximo = p.Proximo;
            return true;
    }
        return false;
       
    }

    public int tamanho()
    {
        throw new NotImplementedException();
    }
    
    public void relatorio(){

       Noh p = null;
       object q = null;

       p = inicio;

       while ( p != null )
       {
       q = p.getItem();
       p = p.Proximo;

       System.Console.Write('\n'+convAluno(q).getNome());
       }
    }

    public Aluno convAluno(object item){

            Aluno r = (Aluno) item;

            return r;
    }
}

File: Prova 2/Mapas/LSE.java
public class LSE {
    
    private Noh inicio;

    public LSE(){
        this.inicio = null;
    }

    public Noh getInicio(){
        return inicio;
    }

    public void insereInicio(Aluno info){
        Noh novo = new Noh(info);

        if(inicio == null){
            inicio = novo;
        } else {
            novo.setProx(inicio);
            inicio = novo;
        }

    }

    public boolean remove(Aluno info){
        Noh p;
        Noh ant = null;

        p = inicio;
        while(p != null && p.getInfo() != info){
            ant = p;
            p = p.getProx();
        }

        if(p == null ){  
            inicio = p.getProx();
            return false;
        }
        
        if(ant == null){
            inicio = p.getProx();
        }
        
        else{
            ant.setProx(p.getProx());
            return true;
    }
        return false;
       
    }

    public void imprimirLista(){

        Noh p = inicio;
        while(p != null){
            System.out.println(p);
            p = p.getProx();
        }
    }
}
